controllers/core/flagsourceconfiguration/kube-flagd-proxy.go

```diff
@@ -0,0 +1,205 @@
+package flagsourceconfiguration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	appsV1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type KubeFlagdProxyHandler struct {
+	client.Client
+	config *KubeProxyConfiguration
+	Log    logr.Logger
+}
+
+type KubeProxyConfiguration struct {
+	Port         int
+	MetricsPort  int
+	DebugLogging bool
+	Image        string
+	Tag          string
+	Namespace    string
+}
+
+func NewKubeFlagdProxyHandler(client client.Client, logger logr.Logger) (*KubeFlagdProxyHandler, error) {
+	kph := &KubeFlagdProxyHandler{
+		config: &KubeProxyConfiguration{
+			Port:         defaultKubeProxyPort,
+			MetricsPort:  defaultKubeProxyMetricsPort,
+			DebugLogging: defaultKubeProxyDebugLogging,
+			Image:        defaultKubeProxyImage,
+			Tag:          defaultKubeProxyTag,
+			Namespace:    defaultKubeProxyNamespace,
+		},
+		Client: client,
+		Log:    logger,
+	}
+	ns, ok := os.LookupEnv(envVarPodNamespace)
+	if ok {
+		kph.config.Namespace = ns
+	}
+	kpi, ok := os.LookupEnv(envVarProxyImage)
+	if ok {
+		kph.config.Image = kpi
+	}
+	kpt, ok := os.LookupEnv(envVarProxyTag)
+	if ok {
+		kph.config.Tag = kpt
+	}
+	portString, ok := os.LookupEnv(envVarProxyPort)
+	if ok {
+		port, err := strconv.Atoi(portString)
+		if err != nil {
+			return kph, fmt.Errorf("could not parse %s env var: %w", envVarProxyPort, err)
+		}
+		kph.config.Port = port
+	}
```

Comment:
since this is duplicated it could be a handy function in the util pkg
something like `getIntEnv(envVar string, default int) (int, error)`